### PR TITLE
Upload endpoint

### LINKED ISSUE #20: Add sad path testing to /upload endpoint
- [ ] Tests all sad paths/edge cases
- [ ] Tests track upload

### LINKED ISSUE #24: Implement AWS SDK and S3 bucket to upload tracks/verses
Implement AWS SDK, implement S3 bucket to add functionality to upload tracks/verses into controller, separate from GraphQL

### PR DESCRIPTION
**Pull request by nmking22**
* Creates ReSTful /upload endpoint to handle verse and track uploads
* Adds testing for happy and sad paths
* All tests passing

Resolves #20
Resolves #24 

## Types of Changes

This PR closes #

- [ ] Bug fix (non-breaking change which fixes an issue): 

- [x] New feature (non-breaking change which adds functionality):

- [ ] Breaking change (fix or feature that would cause existing functionality to change):

- [x] My change requires a change to the documentation.

- [ ] I have updated the documentation accordingly.

- [x] I have added tests to cover my changes.

- [x] All new and existing tests passed.

### CODE REVIEW

File: app/controllers/api/v1/upload_controller.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  end
+
+  def valid_track?
+    (1..12).to_a.include?(params[:month].to_i) && params[:year].present? && params[:type] == 'track'  # && user == admin
+  end
+
+  def invalid_track?
+    params[:type] == 'track'
+  end
+end
```
Comment (the-color-bliu) [inline app/controllers/api/v1/upload_controller.rb:85]: Awesome job with this controller, this is amazing, let's get this refactored into a service as well

File: Gemfile [hunk 2]
```diff
@@ -33,12 +33,18 @@ gem 'faraday'
...
 gem 'graphql'
 
-# Add wrapper for mixcloud api 
+# Add wrapper for mixcloud api
 gem 'mixcloud'
 
+# Allows easier use of AWS S3 bucket interactions
+gem 'aws-sdk-s3', '~> 1'
+
 group :development, :test do
```
Comment (the-color-bliu) [inline Gemfile:48]: Do we need a travis gem?

File: spec/factories.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  factory :verse do
+    track_id { rand(100) }
+    user_id { rand(100) }
+    audio_path { Faker::Internet.url }
+    user
+    track
+  end
+end
```
Comment (the-color-bliu) [inline spec/factories.rb:21]: Love the implementation of factorybot

**Review by the-color-bliu — APPROVED** [review]
Great work, merge approved

File: Gemfile (hunk 2 above)
Comment (ETBassist) [reply Gemfile:48]: I think the Travis gem really just gives us a CLI, but it probably wouldn't hurt to include it

File: spec/factories.rb (hunk 3 above)
Comment (ETBassist) [reply spec/factories.rb:21]: Hahaha, thanks - looking at this now I think I need to specify a range for `month` in the track, I think that sometimes it going to return 0 which isn't a valid month

**Review by ETBassist — APPROVED** [review]
Great says Grommit!
